internal/repository/sync.go

```diff
@@ -64,8 +68,33 @@ func Clone(ctx context.Context, remoteURL, dir, initialBranch string) (*Reposito
 		return nil, errors.Join(ErrCloningRepository, err)
 	}
 
+	sort.Slice(expectedRootKeys, func(i, j int) bool {
+		return expectedRootKeys[i].KeyID < expectedRootKeys[j].KeyID
+	})
+
 	repository := &Repository{r: r}
 
 	slog.Debug("Verifying HEAD...")
+
+	if len(expectedRootKeys) > 0 {
+		state, err := policy.LoadCurrentState(ctx, r)
+		if err != nil {
+			return repository, errors.Join(ErrCloningRepository, err)
+		}
+		rootKeys, err := state.GetRootKeys()
+		if err != nil {
+			return repository, errors.Join(ErrCloningRepository, err)
+		}
+
+		if len(rootKeys) != len(expectedRootKeys) {
+			return repository, ErrClonedAndExpectedKeysDoNotMatch
+		}
+		slog.Debug("Verifying if root keys are expected root keys...")
```

Comment:
I'd move this up before the length check since that's a shortcut to verifying expected root keys.